src/software/geom/spline.cpp

```diff
@@ -0,0 +1,55 @@
+#include "software/geom/spline.h"
+
+Spline::Spline(const std::vector<Point>& points)
+{
+    initLinearSegments(points);
+}
+
+Spline::Spline(const std::initializer_list<Point>& points)
+{
+    initLinearSegments(points);
+}
+
+Point Spline::calculateValue(double val) const
+{
+    if (val < 0)
+    {
+        return start_point;
+    }
+
+    if (val >= segments.size())
+    {
+        return end_point;
+    }
+
+    double x_val = segments[(size_t)val].x.calculateValue(val);
+    double y_val = segments[(size_t)val].y.calculateValue(val);
+    return Point(x_val, y_val);
+}
+
+size_t Spline::size(void) const
+{
+    return segments.size();
+}
+
+void Spline::initLinearSegments(const std::vector<Point>& points)
+{
+    if (points.size() == 1)
```

Comment:
As mentioned above, what if size<1?

Reply:
The member variables should be default initialized to 0 or empty vector. I could be more explicit here...

Reply:
I'd probably prefer an exception in that case. I don't think it makes much sense for a spline with no segments to be a "flat line"?

Reply:
done